src/goto-programs/interpreter.cpp

```diff
@@ -121,147 +272,465 @@ void interpretert::step()
 
   case END_THREAD:
     throw "END_THREAD not yet implemented"; // NOLINT(readability/throw)
     break;
 
   case ATOMIC_BEGIN:
+    trace_step.type=goto_trace_stept::typet::ATOMIC_BEGIN;
     throw "ATOMIC_BEGIN not yet implemented"; // NOLINT(readability/throw)
 
   case ATOMIC_END:
+    trace_step.type=goto_trace_stept::typet::ATOMIC_END;
     throw "ATOMIC_END not yet implemented"; // NOLINT(readability/throw)
 
   case DEAD:
-    throw "DEAD not yet implemented"; // NOLINT(readability/throw)
-
+    trace_step.type=goto_trace_stept::typet::DEAD;
+    break;
+  case THROW:
+    trace_step.type=goto_trace_stept::typet::GOTO;
+    while(!done && (pc->type!=CATCH))
+    {
+      if(pc==function->second.body.instructions.end())
+      {
+        if(call_stack.empty())
+          done=true;
+        else
+        {
+          pc=call_stack.top().return_pc;
+          function=call_stack.top().return_function;
+          call_stack.pop();
+        }
+      }
+      else
+      {
+        next_pc=pc;
+        next_pc++;
+      }
+    }
+    break;
+  case CATCH:
+    break;
   default:
     throw "encountered instruction with undefined instruction type";
   }
-
-  PC=next_PC;
+  pc=next_pc;
 }
 
+/// executes a goto instruction
 void interpretert::execute_goto()
 {
-  if(evaluate_boolean(PC->guard))
+  if(evaluate_boolean(pc->guard))
   {
-    if(PC->targets.empty())
+    if(pc->targets.empty())
       throw "taken goto without target";
 
-    if(PC->targets.size()>=2)
+    if(pc->targets.size()>=2)
       throw "non-deterministic goto encountered";
 
-    next_PC=PC->targets.front();
+    next_pc=pc->targets.front();
   }
 }
 
+/// executes side effects of 'other' instructions
 void interpretert::execute_other()
 {
-  const irep_idt &statement=PC->code.get_statement();
-
+  const irep_idt &statement=pc->code.get_statement();
   if(statement==ID_expression)
   {
-    assert(PC->code.operands().size()==1);
-    std::vector<mp_integer> rhs;
-    evaluate(PC->code.op0(), rhs);
+    assert(pc->code.operands().size()==1);
+    mp_vectort rhs;
+    evaluate(pc->code.op0(), rhs);
+  }
+  else if(statement==ID_array_set)
+  {
+    mp_vectort tmp, rhs;
+    evaluate(pc->code.op1(), tmp);
+    mp_integer address=evaluate_address(pc->code.op0());
+    size_t size=get_size(pc->code.op0().type());
+    while(rhs.size()<size) rhs.insert(rhs.end(), tmp.begin(), tmp.end());
+    if(size!=rhs.size())
+      error() << "!! failed to obtain rhs (" << rhs.size() << " vs. "
+              << size << ")\n" << eom;
+    else
+    {
+      assign(address, rhs);
+    }
+  }
+  else if(statement==ID_output)
+  {
+    return;
   }
   else
     throw "unexpected OTHER statement: "+id2string(statement);
 }
 
 void interpretert::execute_decl()
 {
-  assert(PC->code.get_statement()==ID_decl);
+  assert(pc->code.get_statement()==ID_decl);
 }
 
+/// retrieves the member at offset
+/// \par parameters: an object and a memory offset
```

Comment:
`git grep -w '\\par' | wc -l` yields 273 -- I'll start fixing this, but it looks like a somewhat larger documentation cleanup.